src/constructs/core/mappings.ts

```diff
@@ -1,14 +1,46 @@
 import { CfnMapping } from "@aws-cdk/core";
 import type { Stage } from "../../constants";
+import type { AppIdentity } from "./identity";
 import type { GuStack } from "./stack";
 
-export interface GuStageDependentValue<T extends string | number | boolean> {
+export type GuMappingValue = string | number | boolean;
+
+export interface GuStageMappingValue<T extends GuMappingValue> extends AppIdentity {
   variableName: string;
   stageValues: Record<Stage, T>;
 }
 
-export class GuStageMapping extends CfnMapping {
+export class GuStageMapping {
```

Comment:
Although this class has significant changes, it isn't explicitly tested. Instead we're implicitly testing it - see tests added/updated within this PR.